src/theme/default/components/gridblock/components/variants/DynamicComponentRenderer.tsx

```diff
@@ -10,40 +10,161 @@ interface DynamicComponentRendererProps {
+const isSerializable = (value: any): boolean => {
+  if (value === null || value === undefined) return true;
+  if (typeof value === 'function') return false;
+  if (typeof value === 'object') {
+    try {
+      JSON.stringify(value);
+      return true;
+    } catch (err) {
+      return false;
+    }
+  }
+  return true;
+};
+
 export default function DynamicComponentRenderer({
   item,
   loadedComponent,
   blueprint
 }: DynamicComponentRendererProps) {
-  log.debug(`Rendering component for type: ${item._type} with data:`, loadedComponent?.data);
-
-  const componentProps = {
-    matchingData: loadedComponent.data,
-    componentInformation: {
-      uid: item._key,
-      identifier: item._key,
-      pageDefinition: {
-        source: item._type,
-        pageIdentifier: {
-          identifier: item._key,
-          pageVariant: "subComponentsPage" as PageVariant,
-          isFixedLayout: false
-        }
-      }
-    },
-    blueprint,
-    item
-  };
+  const [isOpen, setIsOpen] = useState(false);
+  
+  const safeItem = useMemo(() => {
+    try {
+      return JSON.parse(JSON.stringify(item));
+    } catch (e) {
+      console.warn("Item không thể serialize, trả về object sạch:", e);
+      return {
+        _type: item._type || 'unknown'
+      };
+    }
+  }, [item]);
+  
+  const safeLoadedComponent = useMemo(() => {
+    if (!loadedComponent) return null;
+    
+    try {
+      return JSON.parse(JSON.stringify(loadedComponent));
+    } catch (e) {
+      console.warn("LoadedComponent không thể serialize, trả về dữ liệu đơn giản:", e);
+      return {
+        data: { _type: item._type || 'unknown' }
+      };
+    }
+  }, [loadedComponent, item._type]);
+  
+  log.debug(`Rendering component for type: ${safeItem._type}`);
 
+  const data = safeLoadedComponent?.data || safeItem;
+  
+  const componentTitle = 
+    data?.title || 
+    data?.heading || 
+    safeItem.title || 
+    safeItem.heading || 
+    `${safeItem._type || 'Unknown'} Component`;
+    
+  const componentDescription = 
+    data?.description || 
+    data?.subtitle ||
+    data?.text ||
+    safeItem.description || 
+    safeItem.text || 
+    safeItem.subtitle || 
+    '';
+  
+  const componentType = safeItem._type?.toLowerCase() || '';
+  
+  const hasBackgroundImage = 
+    (data.backgroundImage && data.backgroundImage.asset && data.backgroundImage.asset.url) ||
+    (safeItem.backgroundImage && safeItem.backgroundImage.asset && safeItem.backgroundImage.asset.url);
+    
+  const backgroundImageUrl = 
+    (data.backgroundImage?.asset?.url) || 
+    (safeItem.backgroundImage?.asset?.url);
+    
+  const button = data.button || safeItem.button;
+  const hasButton = button && button.label;
+  
+  const words = data.words || safeItem.words || data.quote || safeItem.quote;
+  const author = data.author || safeItem.author;
+  
+  const htmlContent = data.content || safeItem.content || data.text || safeItem.text;
+  
+  switch (componentType) {
+    case 'hero':
```

Comment:
![image](https://github.com/user-attachments/assets/46051b75-3319-4743-bcab-14f1683c8173)

Is it possible to avoid having these sub-components in the grid and dyamically render the components on the same level as the gridblock.   To much work as we may have 50 components in a site.